From 6ed7b2a5dda96b1bb60a391a75b5765f2f94fc4d Mon Sep 17 00:00:00 2001
From: Yongxing Mou <quic_yongmou@quicinc.com>
Date: Fri, 24 Jan 2025 16:19:57 +0530
Subject: [PATCH 7/7] BACKPORT: FROMLIST: arm64: dts: qcom: qcs8300-ride:
 Enable Display Port

Enable DPTX0 along with their corresponding PHYs for
qcs8300-ride platform.

Reviewed-by: Konrad Dybcio <konrad.dybcio@oss.qualcomm.com>
Signed-off-by: Yongxing Mou <quic_yongmou@quicinc.com>
Signed-off-by: Mahadevan <quic_mahap@quicinc.com>
Upstream-Status: Submitted [https://lore.kernel.org/all/20250114-dts_qcs8300-v3-2-d114cc5e4af9@quicinc.com/]
---
 arch/arm64/boot/dts/qcom/qcs8300-ride.dts | 44 +++++++++++++++++++++++
 1 file changed, 44 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/qcs8300-ride.dts b/arch/arm64/boot/dts/qcom/qcs8300-ride.dts
index 86ea2dbf29d8..37f971953123 100644
--- a/arch/arm64/boot/dts/qcom/qcs8300-ride.dts
+++ b/arch/arm64/boot/dts/qcom/qcs8300-ride.dts
@@ -38,6 +38,18 @@ sleep_clk: sleep-clk {
 			clock-frequency = <32000>;
 		};
 	};
+
+	dp0-connector {
+		compatible = "dp-connector";
+		label = "DP0";
+		type = "full-size";
+
+		port {
+			dp0_connector_in: endpoint {
+				remote-endpoint = <&mdss_dp0_out>;
+			};
+		};
+	};
 };
 
 &apps_rsc {
@@ -233,6 +245,30 @@ reset-types {
 	};
 };
 
+&mdss {
+	status = "okay";
+};
+
+&mdss_dp0 {
+	pinctrl-0 = <&dp_hot_plug_det>;
+	pinctrl-names = "default";
+
+	status = "okay";
+};
+
+&mdss_dp0_out {
+	data-lanes = <0 1 2 3>;
+	link-frequencies = /bits/ 64 <1620000000 2700000000 5400000000 8100000000>;
+	remote-endpoint = <&dp0_connector_in>;
+};
+
+&mdss_dp0_phy {
+	vdda-phy-supply = <&vreg_l5a>;
+	vdda-pll-supply = <&vreg_l4a>;
+
+	status = "okay";
+};
+
 &qupv3_id_0 {
 	status = "okay";
 };
@@ -257,6 +293,14 @@ &rpmhcc {
 	clock-names = "xo";
 };
 
+&tlmm {
+	dp_hot_plug_det: dp-hot-plug-det-state {
+		pins = "gpio94";
+		function = "edp0_hot";
+		bias-disable;
+	};
+};
+
 &uart7 {
 	status = "okay";
 };
-- 
2.34.1

